src/hermes/model/merge.py

```diff
@@ -131,7 +138,14 @@ def __call__(self, target, path, value, **kwargs):
                 match target[key]:
                     case dict() as item: item.update(value)
                     case list() as item: item[:] = value
-                    case _: target[key] = value
+                    case _:
+                        if target[key] != value:
+                            tag = kwargs.pop('tag', {})
+                            alt = tag.pop('alternatives', [])
+                            alt.append((target[key], tag.copy()))
+                            tag.clear()
+                            tag['alternatives'] = alt
+                        target[key] = value
```

Comment:
**suggestion:** These lines (137-148) are an exact duplicate from lines 99-110. Suggest to extract to a dedicated function or something.